### PR TITLE
Avoid some tree operations

### PR DESCRIPTION
**Pull request by haifeng-jin**
It saves around ~5ms/step in ResNet.

### CODE REVIEW

File: keras_core/layers/layer.py [hunk 1]
```diff
@@ -646,7 +643,15 @@ def maybe_convert(x):
                 return backend.convert_to_tensor(x, dtype=self.compute_dtype)
             return x
 
-        if self._convert_input_args:
+        if (
```
Comment (fchollet) [inline keras_core/layers/layer.py:646]: you can just do `if kwargs or len(args) != 1 or not backend.is_tensor(args[0]) or backend.standardize_dtype(args[0].dtype) != self.compute_dtype: args =...`

File: keras_core/layers/layer.py [hunk 2]
```diff
@@ -646,7 +643,15 @@ def maybe_convert(x):
...
+            not kwargs
+            and len(args) == 1
+            and backend.is_tensor(args[0])
+            and backend.standardize_dtype(args[0].dtype) == self.compute_dtype
+        ):
+            # Used to avoid expensive `tree` operations in the most common case.
+            pass
```
Comment (fchollet) [inline keras_core/layers/layer.py:653]: Basically no need for `if ...: pass; elif:`, a single `if` is enough

**Review by fchollet — COMMENTED** [review]
Thanks for the PR!

File: keras_core/layers/layer.py (hunk 1 above)
Comment (fchollet) [reply keras_core/layers/layer.py:646]: (also I expect a chain of `or` conds to be faster in the negative case because it will exist upon the first match)

Comment (haifeng-jin) [reply keras_core/layers/layer.py:647]: Done.

File: keras_core/layers/layer.py (hunk 2 above)
Comment (haifeng-jin) [reply keras_core/layers/layer.py]: Done.

**Review by fchollet — APPROVED** [review]
LGTM, thank you!

**Pull request merged**